src/HeapDump/Debugger/DumpDebugging/utility.cs

```diff
@@ -100,14 +99,10 @@ public CorDebugPlatform GetPlatform()
         // Implementation of ICorDebugDataTarget.ReadVirtual
         public uint ReadVirtual(ulong address, IntPtr buffer, uint bytesRequested)
         {
-            uint bytesRead = m_reader.ReadPartialMemory(address, buffer, bytesRequested);
-
-            if (bytesRead == 0)
-            {
-                throw new System.Runtime.InteropServices.COMException("Could not read memory requested at address " + address + ".");
-            }
-
-            return bytesRead;
+            uint result = m_reader.ReadPartialMemory(address, buffer, bytesRequested);
+            byte[] data = new byte[bytesRequested];
+            Marshal.Copy(buffer, data, 0, (int)bytesRequested);
+            return result;
```

Comment:
💭 It seems like all of the changes in this file are unnecessary and can be omitted.